nvflare/app_common/executors/launcher_executor.py

```diff
@@ -263,14 +286,28 @@ def _exchange(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abo
                 if reply.data.metrics is not None:
                     self._result_metrics = reply.data
 
-                if self._check_exchange_exit():
-                    break
+            if self._check_exchange_exit():
+                break
+
+            if self._launcher_finish.is_set():
+                self.log_error(
+                    fl_ctx,
+                    f"launcher already exit before exchange ends. exit status is: '{self._launcher_finish_status}'",
```

Comment:
Check this message. Should it be "Launcher already exited before exchange ended."?